introspection-engine/datamodel-renderer/src/datamodel.rs

```diff
@@ -34,6 +34,11 @@ impl<'a> Datamodel<'a> {
         Self::default()
     }
 
+    /// Get the model at the following index.
+    pub fn model_at(&mut self, idx: usize) -> &mut Model<'a> {
+        &mut self.models[idx]
+    }
```

Comment:
This is the controversial bit. I don't see a way around it without a larger refactoring.

Reply:
Alternatively, we could do: `fn push_model_field(&mut self, model_idx: usize, field: Field);` to really avoid consumers of the crate mutating anything we don't want them to.